src/server/pfs/server/driver.go

```diff
@@ -1197,28 +1192,20 @@ func checkPath(path string) error {
 	return nil
 }
 
-func (d *driver) commitExists(commitID string) bool {
-	_, found := d.commitCache.Get(commitID)
-	return found
-}
-
-func (d *driver) setCommitExist(commitID string) {
-	d.commitCache.Add(commitID, struct{}{})
-}
-
 func (d *driver) putFile(ctx context.Context, file *pfs.File, delimiter pfs.Delimiter,
 	targetFileDatums int64, targetFileBytes int64, reader io.Reader) error {
-	// Cache existing commit IDs so we don't hit the database on every
-	// PutFile call.
-	records := &PutFileRecords{}
-	if !d.commitExists(file.Commit.ID) {
-		_, err := d.inspectCommit(ctx, file.Commit)
+	// Check if the commit ID is a branch name.  If so, we have to
+	// get the real commit ID in order to check if the commit does exist
+	// and is open.
+	if len(file.Commit.ID) != uuid.UUIDWithoutDashesLength {
```

Comment:
Isn't this going to fail if someone happens to have a branch that's the same length as a UUID w/o dashes?